src/hooks.js

```diff
@@ -65,29 +68,84 @@ const useResizeListener = ({data, selectedCountries, geolocationState}) => {
   }, [selectedCountries, data, geolocationState])
 }
 
-const useGeolocation = ({fallbackCountry, setGeolocationState, setSelectedCountries, setData}) => {
+const useGeolocation = ({addCountryToStore}) => {
+  const [geolocationState, setGeolocationState] = useState(geolocationStates.requested)
+
+  const getCountrySlug = coords => {
+    const code = countryIso.get(coords.latitude, coords.longitude)
+    return countryData.countries[code].name
+  }
+  
+  const setUpGeolocation = async ({coords, setGeolocationState, addCountryToStore}) => {
+    const slug = getCountrySlug(coords).toLowerCase()
+    setGeolocationState(geolocationStates.loading)
+    addCountryToStore(slug)
+  }
+  
+  const handleGeoError = async ({code, setGeolocationState}) => {
+    const disallowedByUser = code === 1
+    setGeolocationState(
+      disallowedByUser
+        ? geolocationStates.loadingFallback
+        : geolocationStates.failed
+    )
+  }
+
   useEffect(() => {
     if (navigator.geolocation) {
       navigator.geolocation.getCurrentPosition(
-        ({coords}) => setUpGeolocation({coords, setGeolocationState, setSelectedCountries, setData}),
-        ({code}) => handleGeoError({code, fallbackCountry, setGeolocationState, setSelectedCountries, setData})
+        ({coords}) => setUpGeolocation({coords, setGeolocationState, addCountryToStore}),
+        ({code}) => handleGeoError({code, setGeolocationState})
       )
     }
     // eslint-disable-next-line
   }, [])
+
+  return {
+    geolocationState,
+    setGeolocationState
+  }
 }
 
-const useGoogleCharts = () => {
+const useCountrySelectionStore = () => {
+  const [storedCountries, setStoredCountries] = useState([])
+
   useEffect(() => {
-    google.charts.load('current', {
-      packages: ['corechart']
-    })
+    const countries = localStorage.getItem('storedCountries')
```

Comment:
Nit: Maybe call it `selectedCountries` instead. You could prefix literally everything with `stored` in local storage :joy_cat:

Reply:
Sure. Makes sense.